### DISCUSSION TITLE
Human readable config file for keybindings

### DISCUSSION THREAD

**Discussion by normal-max** [mate-desktop/mate-desktop#619] [Ideas]
Hello.
gnome2 appeared at a time when config files were seen as user unfriendly. 
But it seems there is general tendency to not afraid of text config files any more. 
It would be nice if configuration was not limited by gui options and you could change .config/dconf/user by editing.

I really like gnome2 and consider it window XP of linux desktops. It is balanced in many aspects. Not over-designed as many become and not ascetic. Definitely doesn't hate user as gnome3. Has easy readable interface.  It appeared when be sane was important, people were expecting year of Linux desktop :) and MATE bears all userfriendliness of those times. Good option for new users.
But I think there are some good new tendencies worth to catch up with. More advanced users may be more comfortable with editing config files and larger options this may offer.

**Comment by lukefromdc** [comment]
The hard part of this is that some of the gsettings values (those for GTK at a minimum) come from GNOME, who maintain gsettings as a whole. While we could use a different storage system I suppose, we'd have the problem in wayland that all of the wlroots-based compositors that use gsettings at all read GNOME values as those are most likely to be installed. Going forward, as distros talk of "sunsetting xorg" and such, wayland support (we have it now in 1.28)  is likely to grow in importance.

Thus, making these files human-editable would require a collaborative effort across GNOME, MATE, and Cinnamon. Even then there would be backward compatability issues that could force people with existing installs to redo all configuration. Also note that parsing human readable files is slower than loading pure machine readable files. Thus we have the use of gresource to load internal UI files more quickly and as part of the binary. The strongest indicator of this trend is in systemd, where text journal files get partially replaced by journald and its data file.
